test/uk/gov/hmrc/gform/validation/DateValidationSpec.scala

```diff
@@ -16,588 +16,342 @@
-   *
-   *     result.toEither should beLeft(Map(FormComponentId("accPeriodStartDate") -> Set("sample label is missing")))
-   *   }
-   *
-   *   "Date validations" should "return supplied error message" in {
-   *     val date = Date(AnyDate, Offset(0), None)
-   *
-   *     val fieldValue = FormComponent(
-   *       FormComponentId("accPeriodStartDate"),
-   *       date,
-   *       toSmartString("sample label"),
-   *       None,
-   *       None,
-   *       None,
-   *       false,
-   *       false,
-   *       false,
-   *       true,
-   *       false,
-   *       Some(toSmartString("New error message"))
-   *     )
-   *
-   *     val fieldValues = List(fieldValue)
-   *
-   *     val data = mkFormDataRecalculated(
-   *       VariadicFormData.ones(
-   *         FormComponentId("accPeriodStartDate.day")   -> "01",
-   *         FormComponentId("accPeriodStartDate.month") -> "01",
-   *         FormComponentId("accPeriodStartDate.year")  -> "1970"
-   *       ))
-   *
-   *     val result: ValidatedType[Unit] =
-   *       mkComponentsValidator(data).validate(fieldValue, fieldValues, GetEmailCodeFieldMatcher.noop).futureValue
-   *
-   *     result.toEither should beLeft(Map(FormComponentId("accPeriodStartDate") -> Set("New error message")))
-   *   }
-   */
+import cats.data.Validated.{ Invalid, Valid }
+import cats.implicits._
+import java.time.LocalDate
+import munit.FunSuite
+import play.api.i18n.Messages
+import play.api.test.Helpers
+import scala.concurrent.Future
+import uk.gov.hmrc.gform.controllers.CacheData
+import uk.gov.hmrc.gform.fileupload.EnvelopeWithMapping
+import uk.gov.hmrc.gform.graph.FormTemplateBuilder._
+import uk.gov.hmrc.gform.lookup.LookupRegistry
+import uk.gov.hmrc.gform.models.Atom
+import uk.gov.hmrc.gform.models.ids.{ BaseComponentId, IndexedComponentId, ModelComponentId }
+import uk.gov.hmrc.gform.models.{ FormModelSupport, SectionSelectorType, VariadicFormDataSupport }
+import uk.gov.hmrc.gform.models.optics.{ DataOrigin, FormModelVisibilityOptics }
+import uk.gov.hmrc.gform.sharedmodel.VariadicFormData
+import uk.gov.hmrc.gform.sharedmodel.form.{ EnvelopeId, ThirdPartyData }
+import uk.gov.hmrc.gform.sharedmodel.{ LangADT, SmartString, SourceOrigin }
+import uk.gov.hmrc.gform.sharedmodel.formtemplate._
+import uk.gov.hmrc.gform.eval.smartstring.SmartStringEvaluator
+import uk.gov.hmrc.gform.validation.ValidationUtil.ValidatedType
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class DateValidationSpec extends FunSuite with FormModelSupport with VariadicFormDataSupport {
+
+  implicit val l: LangADT = LangADT.En
+
+  private val lookupRegistry = new LookupRegistry(Map.empty)
+
+  implicit val smartStringEvaluator: SmartStringEvaluator = new SmartStringEvaluator {
+    override def apply(s: SmartString, markDown: Boolean): String = s.rawValue(LangADT.En)
+  }
+
+  implicit val messages: Messages = Helpers.stubMessages(
```

Comment:
Minor: Could this be moved to a ExampleMessages trait, with all messages for test. I see this being duplicated in DateValidationLogicSpec

Reply:
That would tight couple those tests. Is that really what we want? In this way every test defined only what it needs, and it even clearly gives a hint what sort of keys are used by a particular test.

Reply:
I know, I had that thought in mind. The messages are just constants and in all cases can represent real messages from the production file. The values themselves have no significance for tests no?

Reply:
> The messages are just constants and in all cases can represent real messages from the production file

Yes.

Values have significance for test, but they can be anything, they don't need to match real message.